feature_engine/encoding/one_hot.py

```diff
@@ -82,6 +86,17 @@ class OneHotEncoder(CategoricalInitMixin, CategoricalMethodsMixin):
         value 0 in all the binary variables. Note that if `top_categories` is not None,
         the parameter `drop_last` is ignored.
 
+        If `top_categories` is being used, `custom_categories` must equal None.
+
+    custom_categories: dict, default=None
+        Accepts a dictionary in which the keys are the variables that the use would like
+        to transform. The keys must match the values of `variables` param.
+
+        The dicitonary values are lists of the categories for each selected variable
+        that are to be one-hot encoded.
+
+        If `custom_categories` is being used, `top_categories` must equal None.
```

Comment:
A few things to think:

- I think custom_categories should override top_categories. So no matter what the user enters, custom_categories takes precedence. It is easier to use like this imo. So here we should say "when custom_categories is entered, top_categories will be ignored.

The other thing to consider is, how do we want custom_categories to interplay with variables. Are we going to allow the user to specify categories for 3 variables for example, while encoding 6? What did you have in mind for this? 

For examples vars a,b,c,d,e are categorical, if variables=None, all will be encoded. If user passes a dictionary with keys, b and c, will variables a,d and e also be encoded? or are we making custom_categories take precedence over variables?

Reply:
Oh, Dr. Che, you never let me get away easy. No wonder I enjoy working w/ you ;)

I'm going to suggest that `variables` is superior. I envision scenarios in which users would like to encode all variables but would like to select which categories are encoded in certain variables.

Consequently, this transformer will require a check when `variables` does not equal `None` and the user utilizes `custom_categories`. The transformer should confirm that the variables included in `custom_categories` are a subset of the variables inputted into the `variables` param.

What do you think? ;)

Reply:
Sounds good.